tests/test-pull-collections.sh

```diff
@@ -0,0 +1,247 @@
+#!/bin/bash
+#
+# Copyright © 2017 Endless Mobile, Inc.
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+# Boston, MA 02111-1307, USA.
+
+set -euo pipefail
+
+. $(dirname $0)/libtest.sh
+
+echo '1..6'
+
+cd ${test_tmpdir}
+
+do_commit() {
+    local repo=$1
+    local branch=$2
+    shift 2
+
+    mkdir -p files
+    pushd files
+    ${CMD_PREFIX} ostree --repo="../${repo}" commit -s "Test ${repo} commit for branch ${branch}" -b "${branch}" --gpg-homedir="${TEST_GPG_KEYHOME}" --gpg-sign="${TEST_GPG_KEYID_1}" "$@" > "../${branch}-checksum"
+    popd
+}
+
+do_summary() {
+    local repo=$1
+    shift 1
+
+    ${CMD_PREFIX} ostree "--repo=${repo}" summary --update --gpg-homedir="${TEST_GPG_KEYHOME}" --gpg-sign="${TEST_GPG_KEYID_1}"
+}
+
+do_collection_ref_show() {
+    local repo=$1
+    local branch=$2
+    shift 2
+
+    echo -n "collection ID: "
+    if ${CMD_PREFIX} ostree "--repo=${repo}" show --print-metadata-key=ostree.collection-binding $(cat "${branch}-checksum")
+    then :
+    else return 1
+    fi
+    echo -n "refs: "
+    if ${CMD_PREFIX} ostree "--repo=${repo}" show --print-metadata-key=ostree.ref-binding $(cat "${branch}-checksum")
+    then return 0
+    else return 1
+    fi
+}
+
+ensure_no_collection_ref() {
+    local repo=$1
+    local branch=$2
+    local error=$3
+    shift 3
+
+    if do_collection_ref_show "${repo}" "${branch}"
+    then
+	assert_not_reached "${error}"
+    fi
+}
+
+do_join() {
+    local IFS=', '
+    echo "$*"
+}
+
+ensure_collection_ref() {
+    local repo=$1
+    local branch=$2
+    local collection_id=$3
+    shift 3
+    local refs=$(do_join "$@")
+
+    do_collection_ref_show "${repo}" "${branch}" >"${branch}-meta"
+    cat "${branch}-meta"
```

Comment:
You can drop this `cat`, since `assert_file_has_content` will print out the content on failure.